### PR TITLE
Add support for Meira

### PR DESCRIPTION
**Pull request by colemarkham**

### CODE REVIEW

**Comment by jackhumbert** [comment]
Awesome! Would you mind updating your readme.md to [match our standard format](https://docs.qmk.fm/documentation_templates.html#keyboard-readmemd-template)? Also, is your RGB code based off of #2090, or is there another source that I'm not aware of? I think we'll want to get these all running off the same code eventually (after merging most likely).

**Comment by colemarkham** [comment]
Oh right thanks, forgot to update the readme.

Yes, that's the same LED controller code. This only does monochrome LEDs though. I modified lighting.c to remove references to RGBLIGHT_ENABLED so that WS2812 RGBs can be used for underglow.

**Comment by jackhumbert** [comment]
Cool! I think it'll make sense to merge the two with some conditionals at some point - I might bug you down the road to do some testing with that :)

File: keyboards/meira/config.h [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+#define BACKLIGHT_LEVELS 10
+#define BACKLIGHT_PWM_MAP {2, 4, 8, 16, 40, 55, 70, 128, 200, 255}
+#define BACKLIGHT_BREATHING
+
+#define RGB_DI_PIN D3
+#define RGBLIGHT_TIMER
+#define RGBLED_NUM 15    // Number of LEDs
+
+#ifdef SUBPROJECT_promicro
```
Comment (jackhumbert) [inline keyboards/meira/config.h]: This format has been replaced with `KEYBOARD_folder_folder`, so `KEYBOARD_meira_promicro` should work here, but this shouldn't be needed - all config.h files are included automatically with the correct declarations :/

File: keyboards/meira/promicro/config.h [hunk 2]
```diff
@@ -0,0 +1,168 @@
+/*
+Copyright 2017 REPLACE_WITH_YOUR_NAME
```
Comment (jackhumbert) [inline keyboards/meira/promicro/config.h]: If you could update these, that would be awesome :)

File: keyboards/meira/rules.mk [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+
+CUSTOM_MATRIX = yes
+
+ifeq ($(strip $(ISSI_ENABLE)), yes)
+    TMK_COMMON_DEFS += -DISSI_ENABLE
+endif
+
+ifeq ($(strip $(WATCHDOG_ENABLE)), yes)
+    TMK_COMMON_DEFS += -DWATCHDOG_ENABLE
+endif
```
Comment (jackhumbert) [inline keyboards/meira/rules.mk]: If you stick a `DEFAULT_FOLDER = meira/promicro` in here, it should let Travis pass it :)

**Comment by colemarkham** [comment]
Thanks for the quick review and catching all my little mistakes.

**Comment by colemarkham** [comment]
> Cool! I think it'll make sense to merge the two with some conditionals at some point - I might bug you down the road to do some testing with that :)
Agreed, I'm happy to help out however you need me to.

**Comment by jackhumbert** [comment]
Thanks!

**Pull request merged**